src/symbolize/gimli/libs_dl_iterate_phdr.rs

```diff
@@ -37,20 +46,24 @@ fn infer_current_exe(base_addr: usize) -> OsString {
 
 /// # Safety
 /// `info` must be a valid pointer.
-/// `vec` must be a valid pointer to `Vec<Library>`
+/// `data` must be a valid pointer to `CallbackData`.
 #[forbid(unsafe_op_in_unsafe_fn)]
 unsafe extern "C" fn callback(
     info: *mut libc::dl_phdr_info,
     _size: libc::size_t,
-    vec: *mut libc::c_void,
+    data: *mut libc::c_void,
 ) -> libc::c_int {
     // SAFETY: We are guaranteed these fields:
     let dlpi_addr = unsafe { (*info).dlpi_addr };
     let dlpi_name = unsafe { (*info).dlpi_name };
     let dlpi_phdr = unsafe { (*info).dlpi_phdr };
     let dlpi_phnum = unsafe { (*info).dlpi_phnum };
     // SAFETY: We assured this.
-    let libs = unsafe { &mut *vec.cast::<Vec<Library>>() };
+    let CallbackData {
+        ret: libs,
+        #[cfg(target_os = "android")]
+        maps,
+    } = unsafe { &mut *data.cast::<CallbackData>() };
```

Comment:
That makes sense to me; I'll try making it unconditional and we'll see if CI is happy.

Reply:
Unfortunately there's still the `cfg(target_os = "hurd")` case, [introduced](https://github.com/rust-lang/backtrace-rs/pull/567/files#diff-94c024636a34c29bad972edeabe708c9fb5d7118f6f34d77d5eed2a21aaf4540) by #567.

I kept the same behaviour of not parsing `/proc/self/maps` at all on Hurd, assuming that might be load bearing somehow.

Reply:
They return a memory map that doesn't conform to the pattern we are looking for.

Reply:
We had a remarkable argument about whether or not the path in a memory map must be an absolute path, or if some other OS might invent another schema. I argued that overcalibrating on how paths look on Linux is incorrect, because other Unix-y OS that otherwise copy Linux-like APIs inevitably will make up new definitions of what a "path" looks like.

I feel Android is pretty much proving me correct here (in that I cannot trust simply treating these things as "paths" to make sense).